pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -191,7 +191,38 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 		originalSnapshot := {{ .GoName }}Snapshot{}
 		currentSnapshot := originalSnapshot.Clone()
 		timer := time.NewTicker(time.Second * 1)
+
+
+		{{- range .Resources}}
+		{{- if not .ClusterScoped }}
+				{{ lower_camel .PluralName }}ByNamespace := make(map[string]{{ .ImportPrefix }}{{ .Name }}List)
```

Comment:
@ilackarms is it safe to assume that watch never returns a nil list, even if it's empty?

Reply:
generated clients return an empty list `var x FooList`
clients implementing the interface can return whatever they want

for what is the context?